src/Card/Card.js

```diff
@@ -1,12 +1,26 @@
+import { useState } from 'react';
+import cl from 'classnames';
 import './Card.scss';
 
-function Card({ className, caption, content }) {
+function Card({ className, caption, content, checkbox, defaultSelected = false }) {
+  const [checkboxState, setCheckboxState] = useState(defaultSelected);
+
+  const onHeaderClick = () => {
+    setCheckboxState(!checkboxState);
+  };
```

Comment:
I think it is redundant functionality. For this moment card should be presentation component.
It is not reason to add state here.